### ISSUE TITLE
Merging Sentry Issues that have linked Jira Issues can result in the Jira link being removed

### ISSUE THREAD

**Issue by jonesphillip** [getsentry/sentry#24291]
## Important Details

What versions of Sentry does this impact?

Saas (sentry.io), Self-hosted

## Description

When manually merging Sentry issues where one has an external linked issue (e.g., Jira issue), the merged issue will sometimes lose the link to the external linked issue.

### Expected Behavior

Merging Sentry issues should retain external linked issue.

**Comment by untitaker** [comment]
This is because issue merge does not use the same list of associated models as other tasks, such as reprocessing or issue deletion. This code: https://github.com/getsentry/sentry/blob/265bd88f3490b79ee650a8175f41306adbb2145c/src/sentry/tasks/merge.py#L91-L102

...should actually use the same models as here:

https://github.com/getsentry/sentry/blob/265bd88f3490b79ee650a8175f41306adbb2145c/src/sentry/deletions/defaults/group.py#L11-L30

...which are also used by reprocessing here:

https://github.com/getsentry/sentry/blob/265bd88f3490b79ee650a8175f41306adbb2145c/src/sentry/reprocessing2.py#L108

I would argue that this is not an ecosystem issue but merge's fault. Now to figure out who owns merge-the-product...

**Comment by untitaker** [comment]
According to internal JIRA comments, this is still an ecosystem issue to some degree because currently there cannot be many JIRA tickets per issue. Meaning that either 1) that needs to change 2) there needs to be some sort of conflict resolution when multiple issues with jira tickets each are merged together.

**Comment by sagor999** [comment]
We have this issue as well and it is very frustrating when this happens. Would love to see this picked up for a fix. 🙏

**Comment by untitaker** [comment]
trying to make sure that this one doesn't entirely fall through the cracks.

**Comment by darioseidl** [comment]
> According to internal JIRA comments, this is still an ecosystem issue to some degree because currently there cannot be many JIRA tickets per issue. Meaning that either 1) that needs to change 2) there needs to be some sort of conflict resolution when multiple issues with jira tickets each are merged together.

Makes sense. When merging Sentry issues with different Jira issues something would need to be done. But the link to Jira can also be lost when merging Sentry issues where only one of them has a Jira issues and the others are empty. In that case it should just keep that one Jira link.
